src/app/plugins/ariana.plugin.ts

```diff
@@ -0,0 +1,79 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, IHttpResponse, Maybe } from '../../common/types';
+
+export class ArianaPlugin extends Plugin {
+  public name: string = 'Ariana Plugin';
+  public description: string = 'Get a random picture of Ariana Grande.';
+  public usage: string = 'ariana';
+  public pluginAlias = ['ari'];
+  public permission: ChannelType = ChannelType.Public;
+
+  private _API_URL: string = 'https://www.gettyimages.com/photos/ariana-grande?page=';
+  private reImgs: RegExp = new RegExp(/<img class="gallery-asset__thumb gallery-mosaic-asset__thumb"(.*?)>/g);
+  private reThumb: RegExp = new RegExp(/https:\/\/(.*?)"/g);
+  private rePages: RegExp = new RegExp(/<span class="PaginationRow-module__lastPage___2pChH">(.*?)<\/span>/g);
+  private rePage: RegExp = new RegExp(/>(.*?)</g);
+  private nPages: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+
+    // get total number of pages of Ariana
+    this.container.httpService
+      .get(this._API_URL + '1')
+      .then((res: IHttpResponse) => {
+        const data: string = res.data;
+        const pagesSearch: Maybe<RegExpMatchArray> = data.match(this.rePages);
```

Comment:
fyi, with array destructuring, we can do something along the following lines:

```
const array = [1,2,3]
const [firstElement] = array; // 1
```

we can translate the same idea to `pagesSearch`